view/frontend/web/insights/add-to-cart-mixin.js

```diff
@@ -0,0 +1,51 @@
+define(['jquery'], function ($) {
+    'use strict';
+
+    var algoliaAddToCartMixin = {
+
+        submitForm: function (form) {
+            if (window.algoliaConfig
+                && algoliaConfig.ccAnalytics.enabled
+                && algoliaConfig.ccAnalytics.conversionAnalyticsMode == 'place_order'
+            ) {
+                this._setQueryIdToForm(form)
+            }
+
+            this._super(form);
+        },
+
+        _setQueryIdToForm: function(form) {
+            if (form.find('button[type="submit"]').length
+                && form.find('button[type="submit"]').data('queryid')) {
+                var queryID = form.find('button[type="submit"]').data('queryid');
+            }
+
+            var queryID = queryID || this._getQueryParamFromCurrentUrl('queryID');
+            if (queryID.length == 0) {
+                return;
+            }
+
+            if (form.find('input[name="queryid"]').length == 0) {
+                form.prepend('<input type="hidden" name="queryID" />');
```

Comment:
you can maybe just directly 

```
form.prepend('<input type="hidden" name="queryID" value="' + queryID +  '" />'); 
```

and skip the next instruction.

Reply:
I think I did this to ensure that if there was already an existing queryID input, that the value would be updated. I'm going to keep as is.